### ISSUE TITLE
"Develop" tutorial fails on first try

### ISSUE THREAD

**Issue by RantyDave** [hashicorp/otto#376]
El Capitan 10.11.2. I haven't actually *tried* anything yet, but I wanted to capture what had actually happened. If you knew the right URL it'd be an easy fix :)

```
DaveRMBP:Downloads dpreece$ mv otto /usr/local/bin/
DaveRMBP:Downloads dpreece$ otto
usage: otto [--version] [--help] <command> [<args>]

Available commands are:
    build      Build the deployable artifact for the app
    compile    Prepares your project for being run.
    deploy     Deploy the application
    dev        Start and manage a development environment
    infra      Builds the infrastructure for the Appfile
    status     Status of the stages of this application
    version    Prints the Otto version

DaveRMBP:Downloads dpreece$ cd ~
DaveRMBP:~ dpreece$ mkdir otto
DaveRMBP:~ dpreece$ cd otto
DaveRMBP:otto dpreece$ git clone https://github.com/hashicorp/otto-getting-started.git
Cloning into 'otto-getting-started'...
remote: Counting objects: 28, done.
remote: Total 28 (delta 0), reused 0 (delta 0), pack-reused 28
Unpacking objects: 100% (28/28), done.
Checking connectivity... done.
DaveRMBP:otto dpreece$ cd otto-getting-started
DaveRMBP:otto-getting-started dpreece$ otto compile
==> Loading Appfile...
==> No Appfile found! Detecting project information...
    No Appfile was found. If there is no Appfile, Otto will do its best
    to detect the type of application this is and set reasonable defaults.
    This is a good way to get started with Otto, but over time we recommend
    writing a real Appfile since this will allow more complex customizations,
    the ability to reference dependencies, versioning, and more.
==> Fetching all Appfile dependencies...
==> Compiling...
    Application:    otto-getting-started (ruby)
    Project:        otto-getting-started
    Infrastructure: aws (simple)

    Compiling infra...
    Compiling foundation: consul
==> Compiling main application...
==> Compilation success!
    This means that Otto is now ready to start a development environment,
    deploy this application, build the supporting infrastructure, and
    more. See the help for more information.
    
    Supporting files to enable Otto to manage your application from
    development to deployment have been placed in the output directory.
    These files can be manually inspected to determine what Otto will do.
DaveRMBP:otto-getting-started dpreece$ otto dev
Would you like Otto to install Vagrant?
  Otto requires vagrant to be installed, but it couldn't be found on your
  system. Otto can install the latest version of vagrant for you. vagrant
  will be installed system-wide since it uses system-specific installers.
  Would you like Otto to install vagrant for you? Alternatively, you may install
  this on your own.
  
  If you answer yes, Otto will install vagrant version 1.8.0.
  
  Please enter 'yes' to continue. Any other value will exit.

  Enter a value: yes

==> Downloading Vagrant v1.8.0...
    URL: https://dl.bintray.com/mitchellh/vagrant/vagrant_1.8.0.dmg

Error building dev environment: Error downloading, status code 404
DaveRMBP:otto-getting-started dpreece$ uname -a
Darwin DaveRMBP.fritz.box 15.2.0 Darwin Kernel Version 15.2.0: Fri Nov 13 19:56:56 PST 2015; root:xnu-3248.20.55~2/RELEASE_X86_64 x86_64

```

**Comment by RantyDave** [comment]
There is no vagrant 1.8.0  .... it runs out at 1.7.4

**Comment by RantyDave** [comment]
Oh.
```
==> Caching SSH credentials from Vagrant...
Error building dev environment: Error executing Vagrant: exit status 1

The error messages from Vagrant are usually very informative.
Please read it carefully and fix any issues it mentions. If
the message isn't clear, please report this to the Otto project.
```

**Comment by bittelc** [comment]
I'm confused by your comment, @RantyDave :

> There is no vagrant 1.8.0 .... it runs out at 1.7.4

Here is [Vagrant version 1.8](https://releases.hashicorp.com/vagrant/1.8.0/).

I'm also getting the same error on when trying to run `otto dev` through the getting started tutorial.

**Comment by leowmjw** [comment]
@RantyDave, @bittelc what hypervisor are you using: Virtualbox, VMWare, lxc?  On my OSX with Virtualbox, running Otto v0.1.2 with Vagrant 1.8; no issues.

Review and checkout the following suggestions of similar cases:
1) A suggestion by @mitchellh  in https://github.com/hashicorp/otto/issues/340:
```
Ah, so the "entered an invalid state" is almost always (I've actually never seen it otherwise) a configuration issue with the underlying hypervisor. In this case, it looks like VirtualBox. The best way to debug this is to start VirtualBox manually (GUI) and manually click "Start" on the VM that Otto created. This should error and VirtualBox sometimes gives you better error messages in the UI.

A common scenario I've seen this is VT-x not enabled in the BIOS.
```

2) Just re-run "otto dev" again - https://github.com/hashicorp/otto/issues/328

3) Just destroy the environment; wait awhile and re-run "otto dev" - https://github.com/hashicorp/otto/issues/325

**Comment by RantyDave** [comment]
> On 30/12/2015, at 4:09 PM, leowmjw <notifications@github.com> wrote:
> 
> @RantyDave <https://github.com/RantyDave>, @bittelc <https://github.com/bittelc> what hypervisor are you using: Virtualbox, VMWare, lxc? On my OSX with Virtualbox, running Otto v0.1.2 with Vagrant 1.8; no issues.
> 
> 
It's not a vagrant problem, otto was attempting to download vagrant 1.8 from a source that only had 1.7.something. I assume this is fixed now and I "lucked out" on that particular server. Mostly I put the issue up so someone could ask whether or not deploying to mitchellh's bintray account was a good idea.

The second issue though...
> 
> Review and checkout the following suggestions of similar cases:
> 1) A suggestion by @mitchellh <https://github.com/mitchellh> in #340 <https://github.com/hashicorp/otto/issues/340>:
> 
> Ah, so the "entered an invalid state" is almost always (I've actually never seen it otherwise) a configuration issue with the underlying hypervisor. In this case, it looks like VirtualBox.

Right. Look what came over smartos-discuss today:

http://lists.omniti.com/pipermail/omnios-discuss/2015-December/006203.html <http://lists.omniti.com/pipermail/omnios-discuss/2015-December/006203.html>

> While running a nightly build of illumos-gate the kernel panicked with
> "BAD TRAP". Running OmniOS r151014 on VMWare Fusion.
and

> We have
> experienced some issues with SmartOS running in VMware Fusion on some
> models of Intel CPU.  I believe there is an erratum about spurious
> page faults when running a hypervisor that makes use of EPT.

I have been getting some odd behaviour from time to time that a soft reboot wasn't fixing. It would seem likely they are related. I was also wondering whether this is a function of VirtualBox running at the same time, mostly accidentally, forgetting it was "part of" docker.

So, yes, maybe VMWare and Intel can have a little chat to discuss what to do next :(

-Dave

**Comment by mitchellh** [comment]
This is fixed in master, 0.2 will come out very soon!
